pack/nodejs/provider.ts

```diff
@@ -5,6 +5,50 @@ import * as pulumi from "@pulumi/pulumi";
 import * as inputApi from "./types/input";
 import * as outputApi from "./types/output";
 
+/**
+ * The provider type for the kubernetes package.
+ */
+export class Provider extends pulumi.ProviderResource {
+    /**
+     * Create a Provider resource with the given unique name, arguments, and options.
+     *
+     * @param name The _unique_ name of the resource.
+     * @param args The arguments to use to populate this resource's properties.
+     * @param opts A bag of options that control this resource's behavior.
+     */
+    constructor(name: string, args: ProviderArgs, opts?: pulumi.ResourceOptions) {
+        let inputs: pulumi.Inputs = {
+            "cluster": args ? args.cluster : undefined,
+            "context": args ? args.context : undefined,
+            "kubeconfig": args ? args.kubeconfig : undefined,
+            "namespace": args ? args.namespace : undefined,
+        };
+        super("kubernetes", name, inputs, opts);
+    }
+}
+
+/**
+ * The set of arguments for constructing a Provider.
+ */
+export interface ProviderArgs {
+    /**
+     * If present, the name of the kubeconfig cluster to use.
+     */
+    readonly cluster?: pulumi.Input<string>;
+    /**
+     * If present, the name of the kubeconfig context to use.
+     */
+    readonly context?: pulumi.Input<string>;
+    /**
+     * The contents of a kubeconfig file. If this is set, this config will be used instead of $KUBECONFIG.
+     */
+    readonly kubeconfig?: pulumi.Input<string>;
```

Comment:
It would be sort of nice if we could type this as `Pulumi.Input<Kubeconfig|string>` and automatically JSON-serialize in the former case. Thoughts?

Reply:
That seems like a good idea.

Reply:
Are the types in `k8s.io/client-go/tools/clientcmd/api` the best source for the canonical `kubeconfig` JSON schema?

Reply:
Hmm, this _should_ be a part of the OpenAPI spec, and therefore we _should_ be generating it, but it's not, and we're not. So I'm not sure if we can generate it....

Reply:
But yes, that is the correct place.

Reply:
Suggest we don't block on that, keep this string for now, and open separate issue to track stronger typing (and generation of kubeconfig schema types).

Reply:
:+1: